### PR TITLE
Bringing Fluent2 Divider to main

### PR DESCRIPTION
**Pull request by sophialee0416**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Bringing Fluent2 Divider to main. 

### Verification
| Before | After | fluent2 |
|---|-----|-----|
| image | image | image |
| image | image | image |
| image | image | image |
| image | image | image |
| image | image | image |
| N/A | image | image |
| N/A | image | image |
| image | image | image |
| image | image | image |
| image | image | image |
| image | image | image |
| image | image | image |

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1343)

### CODE REVIEW

**Comment by harrieshin** [comment]
spot check on dark mode, dark elevated, landscape scenario of tableviewcell which previously had lots of issues?

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/BadgeFieldDemoController.swift [hunk 1]
```diff
@@ -49,6 +49,8 @@ class BadgeFieldDemoController: DemoController {
         badgeField.addBadges(withDataSources: dataSources)
         return badgeField
     }
+
+    private let dividers: [MSFDivider] = (0..<2).map { _ in MSFDivider() }
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/BadgeFieldDemoController.swift:53]: do we need this? can't we just add MSFDivider in line 36 and 40

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PeoplePickerDemoController.swift [hunk 2]
```diff
@@ -56,6 +56,7 @@ final class AsyncImageDemoPersona: PersonaData {
 
 class PeoplePickerDemoController: DemoController {
     var peoplePickers: [PeoplePicker] = []
+    var dividers: [MSFDivider] = []
```
Comment (harrieshin) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/PeoplePickerDemoController.swift:59]: is there particular reason we need this?

File: ios/FluentUI.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -517,6 +525,18 @@
...
+			isa = PBXGroup;
+			children = (
+				0ADBD5862915B9C100C8BC06 /* Divider.swift */,
+				0ADBD5852915B9C100C8BC06 /* DividerModifiers.swift */,
+				0ADBD5832915B9C100C8BC06 /* DividerTokenSet.swift */,
+				0ADBD5842915B9C100C8BC06 /* MSFDivider.swift */,
+			);
+			name = Divider;
+			path = FluentUI/Divider;
+			sourceTree = SOURCE_ROOT;
```
Comment (harrieshin) [inline ios/FluentUI.xcodeproj/project.pbxproj:538]: interesting why this is set to sourceRoot? i would have thought path is "Diver" and sourcetree is "<group>
 like others.

File: ios/FluentUI/Divider/DividerModifiers.swift [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+import UIKit
+
```
Comment (harrieshin) [inline ios/FluentUI/Divider/DividerModifiers.swift:8]: there isn't anything? to modify? why we need this?

**Comment by huwilkes** [comment]
I think the size action might be busted, so could you do a manual size comparison?

File: ios/FluentUI/Divider/DividerModifiers.swift (hunk 4 above)
Comment (huwilkes) [reply ios/FluentUI/Divider/DividerModifiers.swift:8]: https://github.com/microsoft/fluentui-apple/pull/1136#discussion_r939291944

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/PeoplePickerDemoController.swift (hunk 2 above)
Comment (huwilkes) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/PeoplePickerDemoController.swift:59]: https://github.com/microsoft/fluentui-apple/pull/833#discussion_r770239258

Comment (mischreiber) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/PeoplePickerDemoController.swift:59]: No longer needed now that we descend from UIView

File: ios/FluentUI/Divider/DividerModifiers.swift (hunk 4 above)
Comment (sophialee0416) [reply ios/FluentUI/Divider/DividerModifiers.swift:8]: Resolved offline; let's remove it for now and add it back in when the time comes.

File: ios/FluentUI.xcodeproj/project.pbxproj (hunk 3 above)
Comment (sophialee0416) [reply ios/FluentUI.xcodeproj/project.pbxproj]: Yea not sure but I changed it with the next iteration!

**Review by harrieshin — APPROVED** [review]

**Pull request merged**